rust/protocol/src/curve/curve25519.rs

```diff
@@ -1,8 +1,12 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Implementation of the [Curve25519] elliptic curve.
+//!
+//! [Curve25519]: https://en.wikipedia.org/wiki/Curve25519
```

Comment:
This is more "implementation of [X25519](https://en.wikipedia.org/wiki/Curve25519) and [Ed25519](https://en.wikipedia.org/wiki/EdDSA#Ed25519)". `curve25519_dalek` implements the curve operations themselves.

Reply:
Done! I also added a ton of really slick doctests to this module!